app/controllers/users_controller.rb

```diff
@@ -197,6 +197,28 @@ def destroy
     redirect_to(users_path) && return
   end
 
+  def lti_launch_initialize
+    @launch_context = params[:launch_context]
+    # code from show
+    # get courses where user is instructor
+    if current_user.administrator?
+      # if current user is admin, show whatever he requests
+      @cuds = @user.course_user_data
+    else
+      # look for cud in courses where current user is instructor of
+      cuds = @user.course_user_data
+      user_cuds = []
+
+      cuds.each do |cud|
+        next unless cud.instructor?
+
+        user_cud =
+          cud.course.course_user_data.where(user: @user).first
+        user_cuds << user_cud unless user_cud.nil?
```

Comment:
What is the difference between `cud` and `user_cud`?

Reply:
honestly I just copy pasted this code from `show` in `users_controller`, gonna refactor

Reply:
Should be much simpler now